lib/config/Configuration.js

```diff
@@ -0,0 +1,319 @@
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getFrameworkDir() {
+		return this.#frameworkDir;
+	}
+
+	/**
+	 * Directory to store artifacts in.
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getArtifactsDir() {
+		return this.#artifactsDir;
+	}
+
+	/**
+	 * Directory to packages in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getPackagesDir() {
+		return this.#packagesDir;
+	}
+
+	/**
+	 * Directory to store metadata in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getMetadataDir() {
+		return this.#metadataDir;
+	}
+
+	/**
+	 * Directory to store staging artifacts in
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getStagingDir() {
+		return this.#stagingDir;
+	}
+
+	/**
+	 * Lockfiles directory
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getLockDir() {
+		return this.#lockDir;
+	}
+
+	/**
+	 * Cache directory
+	 *
+	 * @public
+	 * @returns {string}
+	 */
+	getCacheDir() {
+		return this.#cacheDir;
+	}
+
+	/**
+	 * @public
+	 * @returns {object} The configuration in a JSON format
+	 */
+	toJSON() {
+		return {
+			cwd: this.#cwd,
+			sources: this.#sources,
+			version: this.#version,
+			cacheMode: this.#cacheMode,
+			ui5HomeDir: this.#ui5HomeDir,
+			snapshotEndpointUrl: this.#snapshotEndpointUrl,
+			frameworkDir: this.#frameworkDir,
+			artifactsDir: this.#artifactsDir,
+			packagesDir: this.#packagesDir,
+			metadataDir: this.#metadataDir,
+			stagingDir: this.#stagingDir,
+			lockDir: this.#lockDir,
+			cacheDir: this.#cacheDir,
+		};
+	}
+}
+
+export default Configuration;
+
+/**
+ * Create Configuration from JSON file
+ *
+ * @public
+ * @static
+ * @param {string} [filePath="~/.ui5rc"] Path to configuration JSON file
+ * @returns {@ui5/project/config/Configuration} Configuration instance
+ */
+export async function fromFile(filePath) {
+	filePath = filePath || path.join(os.homedir(), ".ui5rc");
+
+	const {default: fs} = await import("graceful-fs");
+	const {promisify} = await import("node:util");
+	const readFile = promisify(fs.readFile);
+	let config;
+	try {
+		const fileContent = await readFile(filePath);
+		config = JSON.parse(fileContent);
+	} catch (err) {
+		if (err.code === "ENOENT") { // "File or directory does not exist"
+			config = {};
+		} else {
+			throw err;
+		}
+	}
+	return new Configuration(config);
+}
+
+/**
+ * Save Configuration to a JSON file
+ *
+ * @public
+ * @static
+ * @param {string} [filePath="~/.ui5rc"] Path to configuration JSON file
+ * @param {@ui5/project/config/Configuration} config Configuration to save
+ */
+export async function saveConfig(filePath, config) {
+	filePath = filePath || path.join(os.homedir(), ".ui5rc");
+
+	const {default: fs} = await import("graceful-fs");
+	const {promisify} = await import("node:util");
+	const writeFile = promisify(fs.writeFile);
+
+	return writeFile(filePath, JSON.stringify(config.toJSON()));
+}
+
+
+export async function resolveSnapshotEndpointUrl(settingsXML) {
+	const {default: yesno} = await import("yesno");
```

Comment:
here we introduce a coding relevant only for cli usage... what happens when someone is using the `ui5-project` as a direct dependency?

Reply:
This is a static function, that might be used where needed to resolve the snapshotUrl from settings.xml.
I was indeed wondering where this needs to be positioned, but as it's relevant and at this point, it's a bit too early for the CLI implementation. 
Maybe a flag to enable this check would be sufficient. What do you think?

Reply:
Now it's been moved to a separate change